lib/packmap.js

```diff
@@ -0,0 +1,82 @@
+const fs = require("fs-extra");
+const path = require("path");
+const sortObject = require("sort-object-keys");
+
+module.exports = async function packmap(opts) {
+  const normalizedOutdir = path.normalize(opts.outdir);
+  await fs.remove(normalizedOutdir);
+  await fs.ensureDir(normalizedOutdir);
+  const packageJsonPath = path.resolve(process.cwd(), opts.package);
+  const packageJson = require(packageJsonPath);
+  const importMap = await traversePackage(packageJson);
+  importMap.imports = sortObject(importMap.imports);
+  await fs.appendFile(
+    path.resolve(opts.outdir, "import-map.json"),
+    JSON.stringify(importMap, null, 2)
+  );
+
+  function traversePackage(packageJson, importMap = { imports: {} }) {
+    const dependencies = Object.keys(packageJson.dependencies || {});
+    return Promise.all(
+      dependencies.map(dependency => {
+        if (dependency.startsWith("@types/")) {
+          return;
+        }
+        console.log(`processing dependency ${dependency}`);
+
+        let overrides = {};
+        try {
+          overrides = require(`./package-json-overrides/${dependency}.json`);
+        } catch (err) {
+          // do nothing
+        }
+
+        const packageDir = path.dirname(
+          require.resolve(dependency + "/package.json", {
+            paths: [process.cwd()]
+          })
+        );
+        const depPackageJson = require(path.resolve(
+          packageDir,
+          "package.json"
+        ));
+        Object.assign(depPackageJson, overrides);
+        const libDir =
+          (depPackageJson.directories && depPackageJson.directories.lib) || ".";
+        const dirToCopy = path.resolve(packageDir, libDir);
```

Comment:
Copying the whole directory is probably necessary because there might be subsequent dependencies from the main

Reply:
It copies the whole directory by default, and only copies a sub directory if specified via directories.lib in the package json. See https://docs.npmjs.com/files/package.json#directories for more info